### PR TITLE
Update to API URL generation

### PR DESCRIPTION
**Pull request by froi**
**Summary**

- This change permits the site to consume an API that has not API-KEY. Eg. local dev machine.
- Sets KEY to undefined if no API key is present.
- Only adds api_key to URL when API key is present.

Explain the **motivation** for making this change. What existing problem does the pull request solve?

Users of our code should not have to think about API keys if they are running this project locally against a local API.

This will also help us when dockerizing the app.

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

### CODE REVIEW

File: src/app/services/client/client.service.ts [hunk 1]
```diff
@@ -103,7 +103,12 @@ export class ClientService {
     });
   }
   getAgencies(): Observable<Agency[]> {
-    let url = this.BASE + `agencies?size=1000&api_key=${this.KEY}`;
+    let url = this.BASE + `agencies?size=1000`;
+
+    if(this.KEY) {
+      `&api_key=${this.KEY}`;
```
Comment (DanielJDufour) [inline src/app/services/client/client.service.ts]: @froi , maybe I'm reading this wrong, but shouldn't it be
```url += `&api_key=${this.KEY}`;```

Comment (froi) [reply src/app/services/client/client.service.ts]: Yes. I forgot to put WIP in the title and I missed that.

File: src/app/services/client/client.service.ts [hunk 2]
```diff
@@ -89,8 +89,8 @@ export interface AgencyStatus {
 @Injectable()
 export class ClientService {
 
-  private BASE: string = api.base;
-  private KEY: string = process.env.CODE_GOV_API_KEY || api.key;
+  private BASE: string = api.base || 'http://localhost:3000/';
```
Comment (DanielJDufour) [inline src/app/services/client/client.service.ts]: @froi , I know it's minor, but could you just specify http://localhost:300/ as the base in the [config file]( https://github.com/GSA/code-gov-web/blob/master/config/examples/code-dot-gov.json).  I would prefer to keep as much configuration as possible in one spot.  Everything else looks good.

Comment (froi) [reply src/app/services/client/client.service.ts]: Sure. I do have a recommendation/comment about that. 

I think this needs to be more defensive. There is no validation or "good" feedback for a user that does not include a value in the config.json. The app also behaves very differently when you have some sort of default vs not having a value.

Eg.

Removing or leaving the `base` field blank does not throw any error during build and give a console error that does not really help the user figure out what's going on:

no-default-base

I think we should think a bit on how to make the dev experience "better" and maybe rethink some things about how we are managing our configs in all of our projects. I'll make an issue on code-gov or maybe a new issue on each project to discuss this.

Thanks for the review.

File: src/app/services/client/client.service.ts [hunk 3]
```diff
@@ -89,8 +89,8 @@ export interface AgencyStatus {
 @Injectable()
 export class ClientService {
 
-  private BASE: string = api.base;
-  private KEY: string = process.env.CODE_GOV_API_KEY || api.key;
+  private BASE: string = api.base';
```
Comment (DanielJDufour) [inline src/app/services/client/client.service.ts]: @froi , sorry. little typo here

**Comment by DanielJDufour** [comment]
@froi , I just accepted the SEO PR from @saracope , so this branch is now out of date.  Could you update and let me know when to take another look? Thanks!!

**Pull request merged**